Lib/inspect.py

```diff
@@ -3093,6 +3111,29 @@ def from_callable(cls, obj, *,
                                         follow_wrapper_chains=follow_wrapped,
                                         globals=globals, locals=locals, eval_str=eval_str)
 
+    @classmethod
+    def from_frame(cls, frame):
+        """Constructs Signature from a given frame object."""
+        func_code = frame.f_code
+        pos_count = func_code.co_argcount
+        arg_names = func_code.co_varnames
+        keyword_only_count = func_code.co_kwonlyargcount
+
+        defaults = []
+        for name in arg_names[:pos_count]:
+            if frame.f_locals and name in frame.f_locals:
+                defaults.append(frame.f_locals[name])
```

Comment:
It seems like you can move `if frame.f_locals:` out of this loop and the one below.